p2panda-rs/src/document/document.rs

```diff
@@ -37,32 +39,30 @@ pub(super) fn build_graph(
     Ok(graph)
 }
 
-type FieldKey = String;
 type IsEdited = bool;
 type IsDeleted = bool;
 
 /// Reduce a list of operations into a single view.
-pub(super) fn reduce<T: AsOperation>(
-    ordered_operations: &[T],
-) -> (BTreeMap<FieldKey, OperationValue>, IsEdited, IsDeleted) {
+pub(super) fn reduce(
+    ordered_operations: &[OperationWithMeta],
+) -> (Option<DocumentViewFields>, IsEdited, IsDeleted) {
     let is_edited = ordered_operations.len() > 1;
```

Comment:
I just noticed a problem with this `IsEdited` logic. It will return the `IsEdited` marker for documents that were created and immediately deleted even though such a document has never been edited. I think we should initialise this as `false` and then set it to `true` when encountering the first edit instead.

Reply:
This would also be a good thing to put into the docstring.

Reply:
Yes, nice, good spot!

Here are the handbook changes: https://github.com/p2panda/handbook/pull/170